plugins/coverbrowser.koplugin/bookinfomanager.lua

```diff
@@ -0,0 +1,492 @@
+local Blitbuffer = require("ffi/blitbuffer")
+local DataStorage = require("datastorage")
+local DocumentRegistry = require("document/documentregistry")
+local SQ3 = require("lua-ljsqlite3/init")
+local UIManager = require("ui/uimanager")
+local lfs = require("libs/libkoreader-lfs")
+local logger = require("logger")
+local util = require("ffi/util")
+local splitFilePathName = require("util").splitFilePathName
+local _ = require("gettext")
+
+-- Util functions needed by this plugin, but that may be added to existing base/ffi/ files
+local xutil = require("xutil")
+
+-- Database definition
+local BOOKINFO_DB_VERSION = "1-20170605"
+local BOOKINFO_DB_SCHEMA = [[
+    -- For caching book cover and metadata
+    CREATE TABLE IF NOT EXISTS bookinfo (
+        -- Internal book cache id
+        -- (not to be used to identify a book, it may change for a same book)
+        bcid                INTEGER PRIMARY KEY AUTOINCREMENT,
+
+        -- File location and filename
+        directory           TEXT NOT NULL, -- split by dir/name so we can get all files in a directory
+        filename            TEXT NOT NULL, -- and can implement pruning of no more existing files
```

Comment:
Yes, but you have to think about not forgetting to move the .sdr too on USB.
With this sqlite db, if you move a file or rename a folder, you just have obsolete cache data in sqlite. Nothing really needed to move along. And new cached data will be regenerated on demand.
But that will make for another item in DB management sub menu: Prune obsolete info (better idea for that button title ?)